CRAMv3.tex

```diff
@@ -732,30 +736,67 @@ \subsection{\textbf{Slice header block}}
 
 The slice header block is never compressed (block method=raw). For reference mapped 
 reads the slice header also defines the reference sequence context of the data 
-blocks associated with the slice. Mapped and unmapped reads can be stored within 
-the same slice similarly to BAM file. Slices with unsorted reads must not contain 
-any other types of reads.
+blocks associated with the slice. Mapped reads can be stored along with
+\textbf{placed unmapped}\footnote{Unmapped reads can be \textit{placed} or \textit{unplaced}.
+By placed unmapped read we mean a read that is unmapped according to 
+its BAM bit flags but has position fields filled in, thus "placing"  it on a reference sequence. In contrast, 
+unplaced unmapped reads have have a reference sequence ID of -1 and alignment position of 0.}
+reads on the same reference within the same slice, as is possible in BAM.  
+
+Slices with the Multiple Reference flag (-2) set as the sequence ID in the header may contain reads
+mapped to multiple external references, including unmapped\footnotemark[\value{footnote}] reads (placed on these references or unplaced), 
+but multiple embedded references cannot be combined in this way.  When multiple references are 
+used, the RI data series will be used to determine the reference sequence ID for each record. This 
+data series is not present when only a single reference is used within a slice.
+
+The Unmapped (-1) sequence ID in the header is for slices containing only unplaced 
+unmapped\footnotemark[\value{footnote}] reads.
 
 A slice containing data that does not use the external reference in
 any sequence may set the reference MD5 sum to zero.  This can happen
 because the data is unmapped or the sequence has been stored verbatim
 instead of via reference-differencing.  This latter scenario is
-recommended for unsorted or non-coordinate sorted data.
+recommended for unsorted or non-coordinate-sorted data.
+
+To summarize how slice types affect header fields:
+
+
+\begin{tabular}{|l|l|c|c|}\hline
+ & \textbf{Single Reference} & \textbf{Multiple Reference} & \textbf{Unmapped}
+\tabularnewline
+\hline
+ \textbf{Ref Seq ID} & Ref Seq ID of all reads & -2 & -1
+\tabularnewline
+\hline
+ \textbf{Aln Start} & Min(Aln Start of reads) & 0 / ignored & 0 / ignored
+\tabularnewline
+\hline
+ \textbf{Aln Span} & Span including all reads & 0 / ignored & 0 / ignored
+\tabularnewline
+\hline
+ \textbf{Ref MD5} & Required for external reference;\linebreak{}
+optional for embedded & 16 bytes of \textbackslash{}0 & 16 bytes of \textbackslash{}0
+\tabularnewline
+\hline
+\end{tabular}
```

Comment:
I am unsure how much extra value this table brings, given it is immediately followed by the slice description anyway which basically has the same information in it.  The exception is the MD5 description in the table below which could be made more like this table.  Right now it states:

> MD5 checksum of the reference bases within the slice boundaries or 16 \0 bytes when unused

"When unused" is as clear as mud and I prefer your explanation in the table above.  Can we update the table below to be:

> MD5 checksum of the reference bases within the slice boundaries.  If this slice has reference sequence id of -1 (unmapped) or -2 (multi-ref) the MD5 should be 16 bytes of \textbackslash{}0.   For embedded references, the MD5 can either be all-zeros or the MD5 of the embedded sequence.

Note this last sentence is subtly different to how htsjdk currently works I believe.  If I recall, putting an embedded reference sequence which didn't exactly match the external reference (eg a consensus instead of the genuine reference) demonstrated that the code validated the MD5sum against the external reference and not the embedded reference.  I'm not convinced this is correct behaviour although it does raise the question of whether it is even permitted to have the embedded reference differing to the external reference.

Reply:
OK, updated the text.

I don't have a strong opinion on the new table.

Reply:
I'm a big fan of tables over text, but I agree that the new table doesn't add much as it stands (the original suggestion was to merge it with the table below - but as @jmthibault79 pointed out, that would diverge from the pattern used by other tables in the spec). So I'm fine leaving it out.